src/component/CreateInvoice/index.jsx

```diff
@@ -34,49 +36,101 @@ function CreateInvoice() {
     }, [])
 
     function addDetails() {
-        setDetails(details.push({
+        setDetails(details.concat({
             "quantity": 0,
             "rate": 0,
             "total": 0,
             "description": "Optional text field"
         }))
-        console.log(details)
+    }
+    function minusDetails() {
+        if (details.length > 1) {
+            setDetails(details.slice(0, -1))
+        }
     }
 
-    return (
-        <div className="bg-white">
+    function setRowDetails(newDetails, id) {
+        var newList = details.slice()
+        newList[id] = newDetails
+        setDetails(newList)
+    }
 
-            <Link to="/CreateInvoice">+ New Invoice</Link>
+    function storeClient(id) {
+        setSelectedClient(id)
+    }
 
-            <div className="grid grid-cols-2 place-content-around p-3">
-                <div className="pl-3">
-                    <p className="font-medium pb-2 pt-2">From</p>
-                    <p>Kermit the Frog</p>
-                    <p>Beverly Hills</p>
-                    <p>California</p>
-                    <p className="font-medium pb-2 pt-4">To</p>
-                    <Dropdown clients={clients} />
+    function sendInvoice() {
+        const dataToSend = {
+            "client": selectedClient,
+            "total": details.reduce((carry, detail) => carry + detail.total, 0),
```

Comment:
You're having to calculate the total twice, once here and once in the JSX. Can you find a way to only need this once?